app.js

```diff
@@ -5,14 +5,47 @@ const hbs = require('hbs');
 
 // require spotify-web-api-node package here:
 
+const SpotifyWebApi = require('spotify-web-api-node');
+
 const app = express();
 
 app.set('view engine', 'hbs');
 app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
 
 // setting the spotify-api goes here:
-
+const spotifyApi = new SpotifyWebApi({
+    clientId: process.env.CLIENT_ID,
+    clientSecret: process.env.CLIENT_SECRET
+  });
+  
+  // Retrieve an access token
+  spotifyApi
+    .clientCredentialsGrant()
+    .then(data => spotifyApi.setAccessToken(data.body['access_token']))
+    .catch(error => console.log('Something went wrong when retrieving an access token', error));
 // Our routes go here:
 
+app.get('/', (req, res) => {
+    res.render('home');
+  });
+
+  spotifyApi
```

Comment:
writing code like this means that you are not sure you set the access token before trying to get this data. It is one of the traps of asynchronous programming.